### PR TITLE
fix: reintroduce componentDidMount in JSS manager to address incidental regressions

### PR DESCRIPTION
**Pull request by chrisdholt**
<body>

<footer>
 

For [details on formatting](https://github.com/Microsoft/fast-dna/wiki/pull-request-guidance) pull requests.

### CODE REVIEW

**Comment by nalogiudice** [comment]
Can you link a task/bug so we can understand why this was a regression?

**Comment by chrisdholt** [comment]
> Can you link a task/bug so we can understand why this was a regression?

@nalogiudice I'll provide what detail I can here:

Previously we had a lifecycle method that was being called in `componentWillMount` which is unsafe. In moving to the new lifecycle events, React suggests that you either use the constructor, or use `componentDidMount`.

We moved the logic to update stylesheets to the constructor. Our goal was to ensure that users didn't have a flash of unstyled content (as opposed to doing this work in `componentDidMount`). While helping some projects update to FAST-DNA v3.0 I noticed that styles were all of a sudden broken. While it worked before, all of a sudden issue #774 was suddenly repro'ing. Previously the bug was not manifesting in the repo where the work was being done, and now it was. 

Upon investigation, the cause of this was moving the logic from the lifecycle event to the constructor. The order of constructor vs mount is different. We don't have all the details, but what will likely fix this is the same thing that will resolve #774. For the time being, we can prevent the regression by adding the lifecycle event back in. This gives us the opportunity to come up with a "registry" to ensure that parent components with style overrides/updates/modifications carry more weight than their children.

File: packages/fast-jss-manager-react/src/jss-manager.tsx [hunk 1]
```diff
@@ -81,6 +81,13 @@ export class JSSManager<S, C> extends React.Component<JSSManagerProps<S, C>, JSS
         this.state = state;
     }
 
+    /**
```
Comment (nicholasrice) [inline packages/fast-jss-manager-react/src/jss-manager.tsx]: You can probably comment out line 78 also - that is where the update cycle got moved to.

**Comment by nalogiudice** [comment]
Thanks for the explanation @chrisdholt

**Pull request merged**